met-web/src/components/engagement/view/EmailModal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import { Modal } from '@mui/material';
+import SuccessPanel from './SuccessPanel';
+import FailurePanel from './FailurePanel';
+import EmailPanel from './EmailPanel';
+import TabContext from '@mui/lab/TabContext';
+import TabPanel from '@mui/lab/TabPanel';
+import { EmailModalProps } from './types';
+
+function EmailModal(props: EmailModalProps) {
+    const [formIndex, setFormIndex] = useState('email');
+    const [email, setEmail] = useState('');
+
+    function handleClose() {
+        props.handleClose();
+        setFormIndex('email');
+    }
+
+    function checkEmail() {
+        const filter = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+        if (!filter.test(email)) {
+            setFormIndex('error');
+        } else {
+            setFormIndex('success');
+        }
```

Comment:
can we move this to a common place?like a util

Reply:
Will do that